packages/manager/cypress/e2e/core/cloudpulse/widget-level-dimension-filter.spec.ts

```
@@ -0,0 +1,681 @@
+    mockGetVolumes(mockVolumesEncrypted);
+    mockGetUserPreferences({});
+    // Navigate to the metrics page
+    cy.visitWithLogin('/metrics');
+    cy.wait(['@fetchServices']);
+    cy.wait('@fetchDashboard');
+
+    // Select dashboard and resources
+    ui.autocomplete
+      .findByLabel('Dashboard')
+      .should('be.visible')
+      .type(dashboardName);
+    ui.autocompletePopper
+      .findByTitle(dashboardName)
+      .should('be.visible')
+      .click();
+
+    ui.regionSelect.find().clear();
+    ui.regionSelect.find().click();
+    ui.regionSelect.find().click().type(`${mockRegions[0].label}{enter}`);
+
+    ui.autocomplete
+      .findByLabel('Volumes')
+      .should('be.visible')
+      .type(mockVolumesEncrypted[0].label);
+    ui.autocompletePopper
+      .findByTitle(mockVolumesEncrypted[0].label)
+      .should('be.visible')
+      .click();
+    ui.autocomplete.findByLabel('Volumes').type('{esc}');
+
+    cy.wait(Array(6).fill('@getMetrics'));
+    cy.get('[data-qa-widget]').should('have.length.at.least', 1);
+  });
+
+  it('should verify the widget level dimension filter and validate the API response', () => {
+    const globalFilters = ['region', 'volume_id', 'Port'];
+
+    // Verify tooltip message on hover of filter icon
+    ui.tooltip.findByText('Dimension Filters').should('be.visible');
+    ui.drawer.find().should('not.exist');
+
+    // Open the filter drawer
+    ui.button
+      .findByAttribute(
+        'aria-label',
+        `Widget Dimension Filter${dashboard.widgets[0].label}`
+      )
+      .should('be.visible')
+      .click();
+    cy.get('[data-testid="drawer"]').should('be.visible');
+
+    // Verify drawer content
+    ui.drawer
+      .find()
+      .should('be.visible')
+      .within(() => {
+        cy.get('[data-testid="drawer-title"]')
+          .should('be.visible')
+          .and('contain.text', 'Dimension Filters');
+        cy.get('[data-qa-id="filter-drawer-subtitle"]')
```

The assertion is too loose. contain.text can pass even if the text is partially wrong or polluted by extra content. That weakens the test and hides real failures.cy.get('[data-qa-id="filter-drawer-subtitle"]')
  .should('be.visible')
  .and('have.text', dashboard.widgets[0].label);  also removed all contain.text assertions and replaced them with have.text to enforce exact text matching and avoid false positives from partial or noisy UI content.